### ISSUE TITLE
Failed to enumerate processes: unable to connect to remote frida-server: Connection closed

### ISSUE THREAD

**Issue by RaReMoDz** [frida/frida#1960]
I'm running:
```
adb push frida-server /data/local/tmp/
adb shell
chmod 755 /data/local/tmp/frida-server
/data/local/tmp/frida-server &
```
 
but then I when I try
`frida-ps -U`
I get the following
Failed to enumerate processes: unable to connect to remote frida-server: Connection closed

**Comment by neeleshdatley** [comment]
Try this 
adb shell chmod 777 /data/local/tmp/frida-server
For all 3 files frida-server , burp certificate and injectscript.js  file

**Comment by neeleshdatley** [comment]
Did you Selected the correct architecture of frida server

**Comment by RaReMoDz** [comment]
> Did you Selected the correct architecture of frida server
> Try this adb shell chmod 777 /data/local/tmp/frida-server For all 3 files frida-server , burp certificate and injectscript.js file

Yes I selected the correct architecture, I ran adb shell getprop ro.product.cpu.abilist, which responded with "armeabi-v7a,armeabi" and I downloaded "frida-server-15.1.14-android-arm.xz"

I tried "adb shell chmod 777 /data/local/tmp/frida-server" not sure what you mean by these tho "burp certificate and injectscript.js"

@neeleshdatley

**Comment by neeleshdatley** [comment]
Follow this @RaReMoDz      https://infosecwriteups.com/hail-frida-the-universal-ssl-pinning-bypass-for-android-e9e1d733d29

**Comment by RaReMoDz** [comment]
> Follow this @RaReMoDz https://infosecwriteups.com/hail-frida-the-universal-ssl-pinning-bypass-for-android-e9e1d733d29
> 
> inject js file https://codeshare.frida.re/@pcipolloni/universal-android-ssl-pinning-bypass-with-frida/

I'm not doing ssl pinning tho, I just need the server to start

**Comment by RaReMoDz** [comment]
I'm not doing ssl pinning tho, I just need the server to start
@neeleshdatley

**Comment by neeleshdatley** [comment]
then this would have been worked with that

**Comment by crifan** [comment]
* Simple: use [MagiskFrida](https://github.com/ViRb3/magisk-frida)
* Detail:  https://github.com/frida/frida/issues/92#issuecomment-1655209817

**Comment by mhping** [comment]
In my case, it was because of `adb root`. For some reason, Frida refuses to connect to the server when ADB is running in privileged mode. Start the server, then `adb unroot` and then try `frida-ps`
